src/server/console/linux_virtual_terminal.h

```diff
@@ -86,11 +87,14 @@ class LinuxVirtualTerminal : public ConsoleServices
         std::function<bool()> const& switch_away,
         std::function<bool()> const& switch_back) override;
     void restore() override;
-    std::future<std::unique_ptr<Device>> acquire_device(
+    std::future<std::unique_ptr<mir::Device>> acquire_device(
         int major, int minor,
-        std::unique_ptr<Device::Observer> observer) override;
+        std::unique_ptr<mir::Device::Observer> observer) override;
 
+    class Device;
 private:
+    std::vector<Device*> active_devices;
```

Comment:
Is there any danger that these could be deallocated before they are removed from this vector? If so, could the future hold a `shared_ptr` instead of `unique_ptr`, and allow a `weak_ptr` here?